src/components/POLine/POLineDetails/FieldReceiptStatus.js

```diff
@@ -13,18 +15,33 @@ export const RECEIPT_STATUS = {
   receiptNotRequired: 'Receipt Not Required',
 };
 
+const RECEIPT_STATUSES_BY_ORDER_STATUS = {
+  [PO_WORKFLOW_STATUSES.pending]: [
+    'pending',
+    'receiptNotRequired',
+  ],
+  [PO_WORKFLOW_STATUSES.open]: [
+    'partiallyReceived',
+    'receiptNotRequired',
+    'fullyReceived',
+    'cancelled',
+  ],
+};
+
 class FieldReceiptStatus extends Component {
   render() {
+    const { workflowStatus } = this.props;
+    const statuses = RECEIPT_STATUSES_BY_ORDER_STATUS[workflowStatus] || [];
+
     return (
       <Field
         component={Select}
         label={<FormattedMessage id="ui-orders.poLine.receiptStatus" />}
         name="receiptStatus"
+        disabled={!RECEIPT_STATUSES_BY_ORDER_STATUS[workflowStatus]}
```

Comment:
```suggestion
        disabled={!statuses.length}
```

Reply:
yep, I've thought about it to, and decided to leave as is, but of course way can apply this suggestion